app/com/linkedin/drelephant/mapreduce/fetchers/MapReduceFSFetcherHadoop2.java

```diff
@@ -153,14 +153,17 @@ private DataFiles getHistoryFiles(AnalyticJob job) throws IOException {
 
     // Search files in done dir
     String jobHistoryDirPath = getHistoryDir(job);
-    RemoteIterator<LocatedFileStatus> it = _fs.listFiles(new Path(jobHistoryDirPath), false);
-    while (it.hasNext() && (jobConfPath == null || jobHistPath == null)) {
-      String name = it.next().getPath().getName();
-      if (name.contains(jobId)) {
-        if (name.endsWith("_conf.xml")) {
-          jobConfPath = jobHistoryDirPath + name;
-        } else if (name.endsWith(".jhist")) {
-          jobHistPath = jobHistoryDirPath + name;
+
+    if (_fs.exists(new Path(jobHistoryDirPath))) {
```

Comment:
As discussed offline, if the listfiles is throwing exception causing the job to be dropped, then that means the intermediate path has never been used to analyze the jobs. With this change, it will be. We need to make sure the intermediate folder has the final data before analyzing the job or else we could have wrong values for heuristics. Test for that when you are testing this change.

Reply:
The intermediate files do get analyzed when the finished directory is present but the individual jhist and conf files are absent. In this case, the finished directory itself is not present. This basically checks for that.